src/main/java/frc/robot/UserInterface.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Team 4909, Bionics
+ * Billerica Memorial High School
+ *
+ * Copyright:
+ *   2021 Bionics
+ *
+ * License:
+ *   MIT: https://opensource.org/licenses/MIT
+ *   See the LICENSE file in the project's top-level directory for details.
+ */
+
+package frc.robot;
+
+import java.lang.Object;
+import java.util.HashMap;
+import edu.wpi.first.wpilibj.Joystick;
+import edu.wpi.first.wpilibj2.command.button.JoystickButton;
+import frc.peyton.Drivetrain;
+import frc.bionic.swerve.command.DriveWithJoystick;
+
+public class UserInterface
+{
+  // Registry of objects that our user interface can operate on
+  private static HashMap<String, Object> objectRegistry = new HashMap<String, Object>();
+
+  /**
+   * Add an object to the registry
+   *
+   * @param id
+   *   The ID to use for access to the provided object
+   *
+   * @param obj
+   *   The object accessed via reference by the provided ID
+   */
+  public static void registerObject(String id, Object obj)
+  {
+    objectRegistry.put(id, obj);
+  }
+
+  /**
+   * Create the default user interface. All required objects are expected to
+   * have been registered before this function is called.
+   */
+  public static void createDefaultUI()
+  {
+    createUIJoystick0();
+    createUIJoystick1();
+    createUIDashboard();
+  }
+
+  /**
+   * Create the user interface operated via Joystick 0
+   */
+  private static void createUIJoystick0()
+  {
+    Joystick                  joystick0 = new Joystick(0);
+    Drivetrain                drivetrain = (Drivetrain) objectRegistry.get("Drivetrain");
+
+    if (drivetrain == null)
+    {
+      throw new Error("`Drivetrain` has not been registered with UserInterface");
```

Comment:
I would personally wrap such a null check into a protected getter method so this can be reused across throughout the UI class and is less prone to forgetting such a check.